### PR TITLE
ISPN-7240 Deprecate org.infinispan.commons.util.Base64 and replace usages with java.util.Base64

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-7240

See http://java-performance.info/base64-encoding-and-decoding-performance/ 

NOTE: depends on https://github.com/infinispan/infinispan/pull/4686

### CODE REVIEW

**Comment by gustavonalle** [comment]
CI is sluggish...

**Comment by gustavonalle** [comment]
```ClusterExecutorTest.testExecutorTriConsumerException ``` has a suspect failure, it never failed on master but failed with this PR (and the ickle one), could you take a look?

**Comment by anistor** [comment]
This does not look related. There are only two failures of ClusterExecutorTest in history, one on this PR and one on master a week ago, but that one comes after the whole build started failing due to 'java.lang.OutOfMemoryError: Java heap space' so we can ignore it : http://ci.infinispan.org/viewLog.html?buildId=44910&tab=buildResultsDiv&buildTypeId=bt9

I could not find a failure in the Ickle branch.

**Comment by anistor** [comment]
I've just updated the PR to fix some typos in the commit message.

File: query/src/main/java/org/infinispan/query/impl/DefaultTransformer.java [hunk 1]
```diff
@@ -1,31 +1,64 @@
...
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
 import org.infinispan.query.Transformer;
 
 /**
- * Warning, slow as a dog, uses serialization to get a byte representation of a class.  Implement your own!
- *
- * Repeat. It is HIGHLY RECOMMENDED THAT YOU PROVIDE YOUR OWN IMPLEMENTATION OF {@link org.infinispan.query.Transformer}
+ * WARNING, slow as a senile dog, uses Java Serialization and base64 encoding to get a String representation of an
```
Comment (gustavonalle) [inline query/src/main/java/org/infinispan/query/impl/DefaultTransformer.java]: :smile:

**Comment by gustavonalle** [comment]
@anistor indeed, the failure does not seem related, my mistake

File: core/src/main/java/org/infinispan/persistence/keymappers/WrappedByteArrayOrPrimitiveMapper.java [hunk 2]
```diff
@@ -69,21 +70,20 @@ private String serializeObj(WrappedByteArray mv) throws Exception {
...
     * @return
     * @throws Exception
     */
    private WrappedByteArray deserializeObj(String key) throws Exception {
-      if(externalizer==null)
-         throw new IllegalStateException("Cannot deserialize object: undefined marshaller");
-      byte[] data = Base64.decode(key);
+//      if (externalizer == null)
+//         throw new IllegalStateException("Cannot deserialize object: undefined marshaller");
+      byte[] data = Base64.getDecoder().decode(key);
```
Comment (anistor) [inline core/src/main/java/org/infinispan/persistence/keymappers/WrappedByteArrayOrPrimitiveMapper.java]: Done

**Comment by gustavonalle** [comment]
Integrated, thanks @anistor

**Pull request merged**